staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -213,6 +241,20 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 		return nil, actual, err
 	}
 
+	// Handle strict mode.
+	if s.strict {
+		// Delayed failure for the strict YAML to JSON conversation,
+		// so that the error output also includes GVK.
+		if errorYAMLToJSON != nil {
+			return nil, actual, fmt.Errorf("%#v: %v", actual, errorYAMLToJSON)
+		}
+		// Return an error for unknown JSON fields in strict mode.
+		strictObj := obj.DeepCopyObject()
+		if err := yaml.UnmarshalStrict(data, strictObj); err != nil {
+			return nil, actual, fmt.Errorf("%#v: %v", actual, err)
```

Comment:
I suggest creating a new error type for this or something so the caller can know whether it was the strict checking or something else that failed.